main.cpp

```diff
@@ -68,22 +41,44 @@ namespace Example
 #include <iostream>
 #include <cmath>
 
+//forward declare
+
 struct FloatType;
 struct DoubleType;
 struct IntType;
 
+//Point definition ///////////////////////////////////////////
+
+struct Point
+{
+    Point(float xy) : Point(xy, xy) { }  //This calls the constructor below.
+    Point(float _x, float _y) : x(_x), y(_y) { }
+    Point& multiply(float m);
+    Point& multiply(IntType& m);
+    Point& multiply(FloatType& m);
+    Point& multiply(DoubleType& m);
+    void toString();
+private:
+    Point& multiplyInternal(float m); FIXME this is identical to what 'multiply(float)' used to be.  you gain nothing by adding this.  Make your other multiply functions call 'multiply(float)' and remove this extra function.
+    float x{0}, y{0};
+};
+
 
-/////////////////////////////////////////////////////////////////
+//IntType definition /////////////////////////////////////////////
 
 struct IntType
 {
     IntType(int);
     ~IntType();
     operator int() const { return *intTypeHeap; }
-    IntType& add(int rhs );
-    IntType& subtract(int rhs );
-    IntType& multiply(int rhs );
-    IntType& divide(int rhs );
+    
+    IntType& operator+=( const int other);
+    IntType& operator-=(const int other);
+    IntType& operator/=( const int other);
+    IntType& operator*=(const int other);
+
+    IntType& operator=(const IntType& other);
```

Comment:
the Rule of 3/5/0 hasn't been covered yet.  Your `main()` isn't doing any assignment operations so you have no need for this.
- [ ] Remove this `operator=` function.